opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,18 +12,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# FIXME: Use importlib.metadata when support for 3.11 is dropped if the rest of
-# the supported versions at that time have the same API.
-from importlib_metadata import (  # type: ignore
-    EntryPoint,
-    EntryPoints,
-    entry_points,
-    version,
-)
-
 # The importlib-metadata library has introduced breaking changes before to its
 # API, this module is kept just to act as a layer between the
 # importlib-metadata library and our project if in any case it is necessary to
 # do so.
 
-__all__ = ["entry_points", "version", "EntryPoint", "EntryPoints"]
+from sys import version_info
+
+if version_info < (3, 10):
+
+    # pylint: disable=import-error
+    from importlib_metadata import version  # type: ignore
+    from importlib_metadata import (
+        Distribution,
+        EntryPoint,
+        EntryPoints,
+        PackageNotFoundError,
+        distributions,
+    )
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import requires
+
+else:
+
+    from importlib.metadata import (  # type: ignore
+        Distribution,
+        EntryPoint,
+        EntryPoints,
+        PackageNotFoundError,
+        distributions,
+    )
+    from importlib.metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+
+    from importlib.metadata import requires, version  # isort: skip
+
+
+def entry_points(**params) -> EntryPoints:  # type: ignore
+    """
+    Same as entry_points but requires at least one argument
+
+    For Python 3.8 or 3.9:
```

Comment:
there was a proposal for a unified solution for the `entry_points` API in the [pkg_resources removal PR in the contrib repo](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2181#discussion_r1494303135) which works for all currently supported python versions.
maybe this one could be used here?